src/Visualization/Python/Render1D.py

```diff
@@ -0,0 +1,274 @@
+    '''
+    Renders a single time snap shot
+    :param args: the command line arguments
+    :return: None
+    '''
+    import matplotlib.pyplot as plt
+
+    plt.xlabel("x")
+    plt.ylabel(var_name)
+    plt.title("t = {:.5f}".format(int(time_slice)))
+    plt.plot(coords[int(time_slice)], data[int(time_slice)], 'o')
+    if output_prefix:
+        logging.info("Writing still to file {}.pdf".format(output_prefix))
+        plt.savefig(output_prefix + ".pdf", format='pdf')
+    else:
+        plt.show()
+
+
+def render_animation(var_name, output_prefix, interval, time, coords, data):
+    '''
+    Render an animation of the data
+    :param args: the command line arguments
+    :return: None
+    '''
+    import matplotlib.pyplot as plt
+    import matplotlib.animation as animation
+    fig = plt.figure()
+    ax = plt.axes(xlim=(find_extrema_over_data_set(coords)),
+                  ylim=(find_extrema_over_data_set(data)))
+    line, = ax.plot([], [], 'o', lw=2)
+    ax.set_xlabel('x')
+    ax.set_ylabel(var_name)
+    title = ax.set_title("")
+
+    def init():
+        '''
+        Initialize the animation canvas
+        :return: empty line and title info
+        '''
+        line.set_data([], [])
+        title.set_text("")
+        return line, title
+
+    def animate(iteration):
+        '''
+        Update the animation canvas
+        '''
+        title.set_text("t = {:.5f}".format(time[iteration]))
+        line.set_data(coords[iteration], data[iteration])
+        return line, title
+
+    anim = animation.FuncAnimation(fig,
+                                   animate,
+                                   init_func=init,
+                                   frames=len(time),
+                                   interval=interval)
+    if output_prefix:
+        fps = 1000 / interval
+        logging.info(
+            "Writing animation to file {}.mp4 at {} frames per second".format(
+                output_prefix, fps))
+        anim.save(output_prefix + ".mp4", writer='ffmpeg')
+    else:
+
+        def propagate_exceptions(self, exc, val, tb):
+            raise exc(val).with_traceback(tb)
+
+        # Provides an error message with traceback when an exception is raised
+        tkinter.Tk.report_callback_exception = propagate_exceptions
+        plt.show()
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+    args = parse_cmd_line()
+    if args['output'] is None:
+        mpl.use("TkAgg")
+    else:
+        mpl.use("Agg")
+    if args['file_prefix'] is not None:
+        files = args['file_prefix']
+    else:
+        files = args['filename_list']
+    if args['list_vars']:
+        print_var_names(files)
+        sys.exit(0)
+    time, coords, data = get_data(files, args['var'])
+    if args['interval'] is None:
+        interval = 1000 / args['fps']
+    if args['time'] is None:
+        render_animation(args['var'], args['output'], interval, time, coords,
```

Comment:
This line gives an error if the script is run with `--interval`. Add an `else` clause to the `if args['interval'] is None` where you do this:
```py
interval = args['interval']
```